### PR TITLE
Update transactions constraints

### PR DESCRIPTION
**Pull request by tkhahns**
- For each transaction, if the unique (customer, car) tuple is in the transactions list, no other transaction can be made on that customer or car
- Another transaction can only be done if the previous transaction is either marked as done or deleted

### CODE REVIEW

File: src/test/java/transaction/TransactionListTest.java [hunk 1]
```diff
@@ -313,35 +313,46 @@ void testFindTxsByCustomer() {
...
+                        "Transaction(s) by john doe found:" + System.lineSeparator() +
+                        "[X] TX1 | SAB1234C | John Doe | 5 days" + System.lineSeparator() +
+                        "Start Date: 01-10-2024 | End Date: 06-10-2024" + System.lineSeparator() +
+                        "[ ] TX4 | SGD4091D | John Doe | 2 days" + System.lineSeparator() +
+                        "Start Date: 03-10-2024 | End Date: 05-10-2024" + System.lineSeparator();
 
         String actualOutput = outContent.toString();
 
         // Check that the found transactions are printed
         assertTrue(actualOutput.contains("Transaction(s) by john doe found:"),
```
Comment (rexkoh425) [inline src/test/java/transaction/TransactionListTest.java:348]: LGTM

File: src/main/java/transaction/TransactionList.java [hunk 2]
```diff
@@ -46,12 +56,44 @@ public static void addTxWithoutPrintingInfo(Transaction transaction) {
...
+        }
+        return false;
+    }
+
+    // Helper method to check if a car with a specific license plate already exists in the transaction list
+    private static boolean isCarInTransactionList(String licensePlateNumber) {
+        for (Transaction transaction : transactionList) {
+            if (transaction.getCarLicensePlate().equalsIgnoreCase(licensePlateNumber) && !transaction.isCompleted()) {
+                return true;
+            }
```
Comment (rexkoh425) [inline src/main/java/transaction/TransactionList.java:91]: LGTM

File: src/main/java/exceptions/CarException.java [hunk 3]
```diff
@@ -86,4 +86,9 @@ public static CarException invalidParameters(ArrayList<Integer> errorLines){
         return new CarException(message);
     }
 
+    public static CarException carAlreadyInTransactionList() {
+        String message = "This car has been purchased!";
```
Comment (CT9ARyan) [inline src/main/java/exceptions/CarException.java:90]: Maybe you could change "purchased" to "rented" instead??

**Review by CT9ARyan — COMMENTED** [review]
LGTM 👍

**Comment by AaronZZ10** [comment]
LGTM

**Review by rexkoh425 — APPROVED** [review]

**Pull request merged**